### PR TITLE
DELTA-805 add contexts sdk

### PR DESCRIPTION
**Pull request by MihailTymoshenko**
Replace this text with a description of your proposed change.

[JIRA Ticket](https://bt.smartling.net/browse/DELTA-805)

## Squash Commit Guidelines

This is a reminder on how to format your sqaush commit message. Delete from your PR after noting below.

### Message Header

`feat: add Contexts API`

#### Subject Text

Add Contexts API

### CODE REVIEW

**Review by vbarhatov — CHANGES_REQUESTED** [review]

File: smartling-contexts-api/src/test/java/com/smartling/api/contexts/v2/ContextsApiTest.java [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+    private final static String CREATE_BINDINGS_RESPONSE = "{\n" +
+        "  \"response\": {\n" +
+        "    \"code\": \"SUCCESS\",\n" +
+        "    \"data\": {\n" +
+        "      \"created\": {\n" +
+        "        \"totalCount\": 2,\n" +
+        "        \"items\": [\n" +
+        "        {\n" +
+        "          \"bindingUid\": \"bindingId1\",\n" +
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/test/java/com/smartling/api/contexts/v2/ContextsApiTest.java]: done

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/MatchStatusPTO.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class MatchStatusPTO implements ResponseData
+{
+    private String matchId;
+    private String createdDate;
+    private String modifiedDate;
+    private String status;
+    private String contextId;
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/MatchStatusPTO.java]: done

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/BatchDeleteBindingsRequestPTO.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class BatchDeleteBindingsRequestPTO
+{
+    private List<String> stringHashcodes;
+    private String contentFileUri;
+    private String contextUid;
+    private List<String> bindingUids;
+    private boolean notifyTms;
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/BatchDeleteBindingsRequestPTO.java]: done

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    @POST
+    @Path("/projects/{projectId}/contexts/{contextId}/match/async")
+    MatchIdPTO matchAsync(@PathParam("projectId") String projectId, @PathParam("contextId") String contextId,
+        MatchRequestPTO matchRequestPTO);
+
+    @POST
+    @Path("/projects/{projectId}/contexts/upload-and-match-async")
+    @Consumes (MediaType.MULTIPART_FORM_DATA)
+    MatchIdPTO uploadContextAndMatchAsync(@PathParam("projectId") String projectId,
+        @MultipartForm HtmlContextUploadAndMatchPTO contextUploadPTO);
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java]: fixed

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    BatchBindingPTO createBindings(@PathParam("projectId") String projectId, BindingsRequestPTO bindingsRequest);
+
+    @POST
+    @Path("/projects/{projectId}/bindings/list")
+    @Produces
+    PaginatedListResponse<BindingPTO> listBindings(@PathParam("projectId") String projectId, BatchBindingRequestPTO bindingsRequest, @QueryParam("offset") String offset);
+
+    @DELETE
+    @Path("/projects/{projectId}/bindings/{bindingUid}")
+    void deleteBinding(@PathParam("projectId") String projectId, @PathParam("bindingUid") String bindingUid);
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java]: fixed

**Review by vbarhatov — APPROVED** [review]

**Review by AlexArcher — APPROVED** [review]

**Pull request merged**